### PR TITLE
1657408: Sixth set of 20 new SEO descriptions.

### PR DESCRIPTION
**Pull request by DennisLee-DennisLee**
## Summary

Added the missing description metadata field to 20 articles as part of [User Story 1657408: SEO OKR: [ms.prod(dotnet) ms.technology (dotnet-standard)] Missing description issue (502)](https://dev.azure.com/mseng/TechnicalContent/_workitems/edit/1657408).

### CODE REVIEW

File: docs/standard/asynchronous-programming-patterns/implementing-the-event-based-asynchronous-pattern.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ---
 title: "Implementing the Event-based Asynchronous Pattern"
+description: In this article, learn to implement the Event-based Asynchronous Pattern (EAP) in .NET. EAP is a standard way to package a class having asynchronous features.
```
Comment (tdykstra) [inline docs/standard/asynchronous-programming-patterns/implementing-the-event-based-asynchronous-pattern.md:3]:
```suggestion
description: Learn how to implement the Event-based Asynchronous Pattern (EAP) in .NET. EAP is a standard way to package a class that has asynchronous features.
```

File: docs/standard/attributes/index.md [hunk 2]
```diff
@@ -1,5 +1,6 @@
 ---
 title: "Extending Metadata Using Attributes"
+description: Learn to extend metadata using attributes in .NET. Attributes are keyword-like descriptive declarations to annotate programming elements, like types and fields.
```
Comment (tdykstra) [inline docs/standard/attributes/index.md:3]:
```suggestion
description: Learn how to extend metadata using attributes in .NET. Attributes are keyword-like descriptive declarations to annotate programming elements, like types and fields.
```

File: docs/standard/base-types/regular-expression-example-scanning-for-hrefs.md [hunk 3]
```diff
@@ -1,5 +1,6 @@
 ---
 title: "Regular Expression Example: Scanning for HREFs"
+description: In this article, see an example of regular expressions in .NET. The example searches an input string and displays all href attribute values and their locations.
```
Comment (tdykstra) [inline docs/standard/base-types/regular-expression-example-scanning-for-hrefs.md:3]:
```suggestion
description: See an example of regular expressions in .NET. The example searches an input string and displays all href attribute values and their locations.
```

File: docs/standard/collections/commonly-used-collection-types.md [hunk 4]
```diff
@@ -1,5 +1,6 @@
 ---
 title: "Commonly Used Collection Types"
+description: In this article, learn about commonly used collection types in .NET, such as hash tables, queues, stacks, bags, dictionaries, and lists.
```
Comment (tdykstra) [inline docs/standard/collections/commonly-used-collection-types.md:3]:
```suggestion
description: Learn about commonly used collection types in .NET, such as hash tables, queues, stacks, bags, dictionaries, and lists.
```

File: docs/standard/collections/selecting-a-collection-class.md [hunk 5]
```diff
@@ -1,5 +1,6 @@
 ---
 title: "Selecting a Collection Class"
+description: In this article, learn how to decide which collection class in .NET to choose. Using the wrong type can restrict your use of the collection.
```
Comment (tdykstra) [inline docs/standard/collections/selecting-a-collection-class.md:3]:
```suggestion
description: Learn how to decide which collection class in .NET to choose. Using the wrong type can restrict your use of the collection.
```

File: docs/standard/data/xml/xml-schema-xsd-validation-with-xmlschemaset.md [hunk 6]
```diff
@@ -1,5 +1,6 @@
 ---
 title: "XML Schema (XSD) Validation with XmlSchemaSet"
+description: In this article, learn how to validate XML documents against an XML schema definition language (XSD) schema, using an XmlSchemaSet class in .NET.
```
Comment (tdykstra) [inline docs/standard/data/xml/xml-schema-xsd-validation-with-xmlschemaset.md:3]:
```suggestion
description: Learn how to validate XML documents against an XML schema definition language (XSD) schema, using an XmlSchemaSet class in .NET.
```

File: docs/standard/design-guidelines/member.md [hunk 7]
```diff
@@ -1,5 +1,6 @@
 ---
 title: "Member Design Guidelines"
+description: In this article, learn member design guidelines in .NET. Members include methods, properties, events, constructors, and fields.
```
Comment (tdykstra) [inline docs/standard/design-guidelines/member.md:3]:
```suggestion
description: Learn member design guidelines in .NET. Members include methods, properties, events, constructors, and fields.
```

File: docs/standard/garbage-collection/using-objects.md [hunk 8]
```diff
@@ -1,5 +1,6 @@
 ---
 title: "Using objects that implement IDisposable"
+description: Learn to use objects that implement the IDisposable interface in .NET. Types that use unmanaged resources implement IDisposable to allow resource reclaiming.
```
Comment (tdykstra) [inline docs/standard/garbage-collection/using-objects.md:3]:
```suggestion
description: Learn how to use objects that implement the IDisposable interface in .NET. Types that use unmanaged resources implement IDisposable to allow resource reclaiming.
```

File: docs/standard/io/common-i-o-tasks.md [hunk 9]
```diff
@@ -1,5 +1,6 @@
 ---
 title: "Common I/O Tasks"
+description: In this article, review common I/O tasks. Learn how to do common file tasks & common directory tasks using classes & methods in the System.IO namespace in .NET.
```
Comment (tdykstra) [inline docs/standard/io/common-i-o-tasks.md:3]:
```suggestion
description: Learn how to do common file tasks & common directory tasks using classes & methods in the System.IO namespace in .NET.
```

File: docs/standard/io/handling-io-errors.md [hunk 10]
```diff
@@ -1,5 +1,6 @@
 ---
 title: "Handling I/O errors in .NET"
+description: In this article, learn how to handle I/O errors in .NET. Map error codes to exceptions, handle exceptions in I/O operations, and handle IOException.
```
Comment (tdykstra) [inline docs/standard/io/handling-io-errors.md:3]:
```suggestion
description: Learn how to handle I/O errors in .NET. Map error codes to exceptions, handle exceptions in I/O operations, and handle IOException.
```

File: docs/standard/io/how-to-read-characters-from-a-string.md [hunk 11]
```diff
@@ -1,5 +1,6 @@
 ---
 title: "How to: Read characters from a string"
+description: In this article, learn how to read characters from a string in .NET. See examples of character reading synchronously and asynchronously.
```
Comment (tdykstra) [inline docs/standard/io/how-to-read-characters-from-a-string.md:3]:
```suggestion
description: Learn how to read characters from a string in .NET. See examples of character reading synchronously and asynchronously.
```

File: docs/standard/parallel-programming/how-to-write-a-parallel-foreach-loop-with-partition-local-variables.md [hunk 12]
```diff
@@ -1,5 +1,6 @@
 ---
 title: "How to: Write a Parallel.ForEach loop with partition-local variables"
+description: In this article, see an example of how to write a Parallel.ForEach loop that uses partition-local variables in .NET.
```
Comment (tdykstra) [inline docs/standard/parallel-programming/how-to-write-a-parallel-foreach-loop-with-partition-local-variables.md:3]:
```suggestion
description: See an example of how to write a Parallel.ForEach loop that uses partition-local variables in .NET.
```

File: docs/standard/security/how-to-sign-xml-documents-with-digital-signatures.md [hunk 13]
```diff
@@ -1,5 +1,6 @@
 ---
 title: "How to: Sign XML Documents with Digital Signatures"
+description: In this article, learn how to sign XML documents with digital signatures. Use classes in the System.Security.Cryptography.Xml namespace in .NET.
```
Comment (tdykstra) [inline docs/standard/security/how-to-sign-xml-documents-with-digital-signatures.md:3]:
```suggestion
description: Learn how to sign XML documents with digital signatures. Use classes in the System.Security.Cryptography.Xml namespace in .NET.
```

File: docs/standard/security/how-to-use-data-protection.md [hunk 14]
```diff
@@ -1,5 +1,6 @@
 ---
 title: "How to: Use Data Protection"
+description: In this article, learn how to use data protection by accessing the data protection API (DPAPI) in .NET.
```
Comment (tdykstra) [inline docs/standard/security/how-to-use-data-protection.md:3]:
```suggestion
description: Learn how to use data protection by accessing the data protection API (DPAPI) in .NET.
```

File: docs/standard/security/walkthrough-creating-a-cryptographic-application.md [hunk 15]
```diff
@@ -1,5 +1,6 @@
 ---
 title: "Walkthrough: Creating a Cryptographic Application"
+description: In this article, walk through the creation of a cryptographic application. Learn how to encrypt and decrypt content in a Windows Forms application.
```
Comment (tdykstra) [inline docs/standard/security/walkthrough-creating-a-cryptographic-application.md:3]:
```suggestion
description: Walk through the creation of a cryptographic application. Learn how to encrypt and decrypt content in a Windows Forms application.
```

**Review by tdykstra — APPROVED** [review]
These look great.  I made some suggestions, mainly removing "In this article" at the start of each description.

**Comment by DennisLee-DennisLee** [comment]
Thanks, @tdykstra. I've accepted your suggestions.

Many of the SEO description examples in https://review.docs.microsoft.com/en-us/help/contribute/contribute-how-to-write-seo-meta-description?branch=master use the introductory clauses you wanted to get rid of. I didn't really think that type of throat-clearing was needed, but I went along. I'll abide by your request next time.

#sign-off

**Pull request merged**